src/main/java/aitsi/m3spin/tests/AstImplTest.java

```diff
@@ -0,0 +1,199 @@
+package aitsi.m3spin.tests;
+
+import aitsi.m3spin.commons.enums.EntityType;
+import aitsi.m3spin.commons.enums.LinkType;
+import aitsi.m3spin.commons.impl.AssignmentImpl;
+import aitsi.m3spin.commons.impl.StatementImpl;
+import aitsi.m3spin.commons.impl.VariableImpl;
+import aitsi.m3spin.commons.interfaces.Assignment;
+import aitsi.m3spin.commons.interfaces.Statement;
+import aitsi.m3spin.commons.interfaces.Variable;
+import aitsi.m3spin.pkb.exception.IllegalNodeTypeException;
+import aitsi.m3spin.pkb.impl.AstImpl;
+import aitsi.m3spin.pkb.interfaces.Ast;
+import org.junit.jupiter.api.Test;
+import org.opentest4j.TestAbortedException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AstImplTest {
+
+    private final Ast ast = new AstImpl();
+
+    @Test
+    void createTNode() {
+
+        try
+        {
+            assertAll(
+                    () -> assertEquals(EntityType.ASSIGNMENT, ast.createTNode(EntityType.ASSIGNMENT).getType()),
+                    () -> assertEquals(EntityType.CONSTANT, ast.createTNode(EntityType.CONSTANT).getType()),
+                    () -> assertEquals(EntityType.IF, ast.createTNode(EntityType.IF).getType()),
+                    () -> assertEquals(EntityType.MINUS, ast.createTNode(EntityType.MINUS).getType()),
+                    () -> assertEquals(EntityType.PLUS, ast.createTNode(EntityType.PLUS).getType()),
+                    () -> assertEquals(EntityType.PROCEDURE, ast.createTNode(EntityType.PROCEDURE).getType()),
+                    () -> assertEquals(EntityType.VARIABLE, ast.createTNode(EntityType.VARIABLE).getType()),
+                    () -> assertEquals(EntityType.WHILE, ast.createTNode(EntityType.WHILE).getType()),
+                    () -> assertThrows(IllegalNodeTypeException.class, () -> ast.createTNode(EntityType.EXPRESSION)),
+                    () -> assertThrows(IllegalNodeTypeException.class, () -> ast.createTNode(EntityType.STMT)),
+                    () -> assertThrows(IllegalNodeTypeException.class, () -> ast.createTNode(EntityType.STMT_LIST))
+            );
+        }
+        catch (Exception e) {
+            throw new TestAbortedException();
+        }
+
+    }
+
+    @Test
+    void setRoot() {
+        Statement root = new StatementImpl();
+        ast.setRoot(root);
+
+        assertEquals(ast.getRoot(), root);
```

Comment:
Powinno być:
expected value: zmienna `root`
actual value: to co zwróci `setRoot()`